src/toxcore/onion/onion_client.rs

```diff
@@ -292,6 +294,25 @@ impl OnionClient {
         }
     }
 
+    fn is_redundant_ping(&self, pk: PublicKey, search_pk: PublicKey, request_queue: &RequestQueue<AnnounceRequestData>) -> bool {
+        let check_pks = |data: &AnnounceRequestData| -> bool {
+            let request_search_pk = if let Some(friend_pk) = data.friend_pk {
+                friend_pk
+            } else {
+                self.dht.pk
+            };
+            data.pk == pk && search_pk == request_search_pk
+        };
+        let request = request_queue.get_values()
+            .filter(|(ping_time, _request_data)| clock_elapsed(*ping_time) <= request_queue.get_timeout())
```

Comment:
Wrong place to filter timed out records. It doesn't make sense just because next you check time anyway. It rather should be [here](https://github.com/tox-rs/tox/pull/325/files#diff-6e41041a9b7f108d886020985a2290c2R57) to make this function consistent regardless of `clear_timed_out` execution.

Reply:
Why?